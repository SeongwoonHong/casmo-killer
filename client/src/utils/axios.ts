import axios from 'axios';

axios.interceptors.request.use((config) => {
  console.log(config);
  return config;
});

function setTokenToHeader(header, csrfToken) {
  axios.defaults.headers.common[header] = csrfToken;
}

export {
  axios,
  setTokenToHeader,
};
