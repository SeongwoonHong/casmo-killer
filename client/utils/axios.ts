import axios from 'axios';

const baseURL = 'http://localhost:9000';

axios.defaults.baseURL = baseURL + '/api';

export { axios };
