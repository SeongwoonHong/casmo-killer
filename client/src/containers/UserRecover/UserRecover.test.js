/* eslint react/jsx-boolean-value: 0 */
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import axios from 'axios';

import UserRecover from './UserRecover';

describe('<UserRecover />', () => {

  let component;
  let mockPost;

  beforeEach(() => {
    component = shallow(
      <UserRecover />
    );
    mockPost = sinon.stub(axios, 'post');
  });

  afterEach(() => {
    mockPost.restore();
  });

  it('should render the component consistently', () => {
    expect(component).toMatchSnapshot();
  });

  it('should construct the component with the correct state', () => {
    expect(component.state()).toEqual({
      isLoading: false,
      isSuccess: false,
      email: '',
      message: ''
    });
  });

  it('should make sure that value is entered for email', () => {

    component.setState({ email: '' });

    component.instance().onSubmitHandler();

    expect(component.state('isSuccess')).toEqual(false);
    expect(component.state('message')).toEqual('Please enter your email address');

  });

  it('should send the data to the server and resolve the response to the state', async () => {

    const responseData = { data: { message: 'server message' } };
    const responsePromise = Promise.resolve(responseData);

    mockPost.callsFake(() => responsePromise);

    component.setState({ email: 'ckboard@gmail.com' });

    component.instance().onSubmitHandler();

    await responsePromise;

    expect(component.state('isSuccess')).toEqual(true);
    expect(component.state('message')).toEqual(responseData.data.message);

  });

});
