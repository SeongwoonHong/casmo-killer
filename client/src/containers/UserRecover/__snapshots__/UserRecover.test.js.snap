// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserRecover /> should render the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Recover />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "button": <button
        className="User__form__btn"
        type="submit"
>
        Submit
</button>,
      "children": Array [
        <FormMessage
          message=""
          type="error"
/>,
        <UserInputField
          disabled={false}
          isVisible={true}
          message="The instruction email will be sent to this email."
          name="email"
          onChange={[Function]}
          placeholder=""
          title=""
          type=""
          value=""
/>,
      ],
      "className": "Recover",
      "disabled": false,
      "formMsg": "Enter your email address below, and we will send you a link where you can reset your password.",
      "formTitle": "Please enter your email",
      "icon": "refresh",
      "isLoading": false,
      "onSubmit": [Function],
      "title": "Reset Your Password",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "message": "",
          "type": "error",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": null,
          "disabled": false,
          "isVisible": true,
          "message": "The instruction email will be sent to this email.",
          "name": "email",
          "onChange": [Function],
          "placeholder": "",
          "title": "",
          "type": "",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "button": <button
          className="User__form__btn"
          type="submit"
>
          Submit
</button>,
        "children": Array [
          <FormMessage
            message=""
            type="error"
/>,
          <UserInputField
            disabled={false}
            isVisible={true}
            message="The instruction email will be sent to this email."
            name="email"
            onChange={[Function]}
            placeholder=""
            title=""
            type=""
            value=""
/>,
        ],
        "className": "Recover",
        "disabled": false,
        "formMsg": "Enter your email address below, and we will send you a link where you can reset your password.",
        "formTitle": "Please enter your email",
        "icon": "refresh",
        "isLoading": false,
        "onSubmit": [Function],
        "title": "Reset Your Password",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "message": "",
            "type": "error",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": null,
            "disabled": false,
            "isVisible": true,
            "message": "The instruction email will be sent to this email.",
            "name": "email",
            "onChange": [Function],
            "placeholder": "",
            "title": "",
            "type": "",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
